tests/integration/test_subproject_creator.py

```diff
@@ -0,0 +1,152 @@
+import os
+import subprocess
+from pathlib import Path
+
+import yaml
+
+from dbt_meshify.dbt import Dbt
+from dbt_meshify.dbt_projects import DbtProject
+from dbt_meshify.storage.dbt_project_creator import DbtSubprojectCreator
+from dbt_meshify.storage.file_content_editors import DbtMeshConstructor
+
+test_project_profile = yaml.safe_load(
+    """
+test:
+  outputs:
+    dev:
+      path: ./database.db
+      threads: 2
+      type: duckdb
+  target: dev
+"""
+)
+
+test_package_yml = yaml.safe_load(
+    """
+packages:
+  - package: dbt-labs/dbt_utils
+    version: 1.0.0
+"""
+)
+
+model_unique_id = "model.test.my_first_dbt_model"
+
+
+def setup_new_project(write_packages_yml: bool = False):
+    dbt = Dbt()
+    subprocess.run(["poetry", "run", "dbt", "init", "test", "-s"])
+    with open("test/profiles.yml", "w") as f:
+        f.write(yaml.dump(test_project_profile))
+    if write_packages_yml:
+        with open("test/packages.yml", "w") as f:
+            f.write(yaml.dump(test_package_yml))
+        dbt.invoke(directory=Path("test"), runner_args=["deps"])
+
+
+def split_project(select: str = "my_first_dbt_model"):
+    project = DbtProject.from_directory(Path("test"))
+    subproject = project.split(project_name='subdir', select=select)
+    return subproject
+
+
+def get_meshify_constructor(subproject, unique_id):
+    resource = subproject.get_manifest_node(unique_id)
+    if not resource:
+        raise KeyError(f"Resource {unique_id} not found in manifest")
+    meshify_constructor = DbtMeshConstructor(
+        project_path=subproject.path, node=resource, catalog=None
+    )
+    return meshify_constructor
+
+
+def teardown_new_project():
+    os.system("rm -rf test-projects/test")
```

Comment:
😉 let's use `shutil.rmtree` instead of `os.system`.

More generally, it is now recommended to use `subprocess.run` rather than `os.system` when running commands from Python